server/rosetta/cosmos/client/client.go

```diff
@@ -0,0 +1,562 @@
+		for _, addr := range addrs {
+			signer := &types.AccountIdentifier{
+				Address: addr.String(),
+			}
+			accountIdentifierSigners = append(accountIdentifierSigners, signer)
+		}
+	}
+
+	return conversion.SdkTxToOperations(txBldr.GetTx(), false, false), accountIdentifierSigners, nil
+}
+
+// GetTx returns a transaction given its hash
+func (c *Client) GetTx(_ context.Context, hash string) (*types.Transaction, error) {
+	txResp, err := authclient.QueryTx(c.clientCtx, hash)
+	if err != nil {
+		return nil, rosetta.WrapError(rosetta.ErrUnknown, err.Error())
+	}
+	var sdkTx sdk.Tx
+	err = c.ir.UnpackAny(txResp.Tx, &sdkTx)
+	if err != nil {
+		return nil, rosetta.WrapError(rosetta.ErrCodec, err.Error())
+	}
+	return conversion.SdkTxWithHashToRosettaTx(&rosetta.SdkTxWithHash{
+		HexHash: txResp.TxHash,
+		Code:    txResp.Code,
+		Log:     txResp.RawLog,
+		Tx:      sdkTx,
+	}), nil
+}
+
+// GetUnconfirmedTx gets an unconfirmed transaction given its hash
+// NOTE(fdymylja): not implemented yet
+func (c *Client) GetUnconfirmedTx(_ context.Context, _ string) (*types.Transaction, error) {
+	return nil, rosetta.WrapError(rosetta.ErrNotImplemented, "get unconfirmed transaction method is not supported")
+}
+
+// Mempool returns the unconfirmed transactions in the mempool
+func (c *Client) Mempool(ctx context.Context) ([]*types.TransactionIdentifier, error) {
+	txs, err := c.clientCtx.Client.UnconfirmedTxs(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return conversion.TMTxsToRosettaTxsIdentifiers(txs.Txs), nil
+}
+
+// Peers gets the number of peers
+func (c *Client) Peers(ctx context.Context) ([]*types.Peer, error) {
+	netInfo, err := c.clientCtx.Client.NetInfo(ctx)
+	if err != nil {
+		return nil, rosetta.WrapError(rosetta.ErrUnknown, err.Error())
+	}
+	return conversion.TmPeersToRosettaPeers(netInfo.Peers), nil
+}
+
+func (c *Client) Status(ctx context.Context) (*types.SyncStatus, error) {
+	status, err := c.clientCtx.Client.Status(ctx)
+	if err != nil {
+		return nil, rosetta.WrapError(rosetta.ErrUnknown, err.Error())
+	}
+	return conversion.TMStatusToRosettaSyncStatus(status), err
+}
+
+func (c *Client) getTxConfig() client.TxConfig {
+	return c.clientCtx.TxConfig
+}
+
+func (c *Client) PostTx(txBytes []byte) (*types.TransactionIdentifier, map[string]interface{}, error) {
+	res, err := c.clientCtx.BroadcastTx(txBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &types.TransactionIdentifier{
+			Hash: res.TxHash,
+		},
+		map[string]interface{}{
+			rosetta.Log: res.RawLog,
+		}, nil
+}
+
+func (c *Client) SignedTx(ctx context.Context, txBytes []byte, signatures []*types.Signature) (signedTxBytes []byte, err error) {
+	TxConfig := c.getTxConfig()
+	rawTx, err := TxConfig.TxDecoder()(txBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	txBldr, _ := TxConfig.WrapTxBuilder(rawTx)
+
+	var sigs = make([]signing.SignatureV2, len(signatures))
+	for i, signature := range signatures {
+		if signature.PublicKey.CurveType != "secp256k1" {
```

Comment:
ah this is certainly a problem for chains like Ethermint, why is the secp256k1 the only supported key? cc @alessio

Reply:
IIRC wasn't it due to the sdk auth module supporting only secp256k1 for signing?